src/core/obj.js

```diff
@@ -177,6 +177,48 @@ class Catalog {
     return (root.items.length > 0 ? root.items : null);
   }
 
+  get permissions() {
+    let permissions = null;
+    try {
+      permissions = this._readPermissions();
+    } catch (ex) {
+      if (ex instanceof MissingDataException) {
+        throw ex;
+      }
+      warn('Unable to read permissions.');
+    }
+    return shadow(this, 'permissions', permissions);
+  }
+
+  /**
+   * @private
+   */
+  _readPermissions() {
+    const encrypt = this.xref.trailer.get('Encrypt');
+    if (!isDict(encrypt)) {
+      return null;
+    }
+
+    let flags = encrypt.get('P');
+    if (!isNum(flags)) {
+      return null;
+    }
+
+    // PDF integer objects are represented internally in signed 2's complement
+    // form. Therefore, convert the signed decimal integer to a signed 2's
+    // complement binary integer so we can use regular bitwise operations on it.
+    flags += 2 ** 32;
+
+    const permissions = [];
+    for (const key in PermissionFlag) {
+      const value = PermissionFlag[key];
```

Comment:
If you include something along these lines https://github.com/mozilla/pdf.js/pull/9982/commits/e59f952a59315a7346cc0fc7277c9067508c479f, then you ought to be able to use `for...of` instead here:
```javascript
for (const flag of PermissionFlag) {
  if (flags & flag) {
    ...
}
```

Reply:
I tried this and unfortunately it doesn't work. Objects, unlike other data structures such as `Map`s, are not iterable in JavaScript. According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/is_not_iterable there are some other ways to do this but I don't think it's much more readable.

Reply:
I totally overlooked that it's an Object, sorry about wasting your time here.